src/Language/DifferentialDatalog/Validate.hs

```diff
@@ -80,22 +81,27 @@ validate d = do
                  }
     -- Validate dependency graph
     depGraphValidate d''
+    {-
     -- This check must be done after 'depGraphValidate', which may
     -- introduce recursion
     checkNoRecursion d''
+    -}
     -- Attributes do not affect the semantics of the program and can therefore
     -- be validated last.
     progValidateAttributes d''
     return d''
 
+{-
```

Comment:
are you still doing this someplace else?

Reply:
no, recursive functions are now allowed